Film project/app.js

```diff
@@ -0,0 +1,131 @@
+const apiKey = '0fcff4e02ae310d6aa49e8a3396e69cd';
+const baseUrl = 'https://api.themoviedb.org/3';
+
+const getByActorForm = document.querySelector('#getbyactorForm');
+const responseContainer = document.querySelector('.response-container');
+const getByActorFormInput = document.querySelector('.actorInput');
+const formContainer = document.querySelectorAll('.form-container');
+const getByYearForm = document.querySelector('#getbyyearForm');
+const getByYearFormInput = document.querySelector('.year-input');
+const getByGenreForm = document.querySelector('#getbygenreForm');
+const selectEl = document.querySelector('#genres');
+
+
+
+formContainer.forEach(container => container.addEventListener('click', function (e) {
+    document.querySelectorAll('.form-area--hidden').forEach(area => area.hidden = true);
+    const form = e.currentTarget.firstElementChild;
+    const { id } = e.currentTarget;
+    const areaToShow = form.querySelector(`[aria-labelledby="${id}"]`);
+    areaToShow.hidden = false;
+}));
+
+getByActorForm.addEventListener('submit', function (e) {
+    e.preventDefault();
+    getPerson();
+});
+
+function getPerson() {
+    fetch(`${baseUrl}/search/person?api_key=${apiKey}&query=${getByActorFormInput.value}`)
+        .then(response => response.json())
+        .then(data => {
+            if (data && data.results && data.results.length > 0) {
+                getMovieListByPersonId(data.results[0].id);
+            } else {
+                responseContainer.innerHTML = '<p class="error-message">No results.</p>'
+            }
+        })
+        .catch(err => console.log(err));
+}
+
+function getMovieListByPersonId(id) {
+    fetch(`${baseUrl}/discover/movie?api_key=${apiKey}&with_people=${id}`)
+        .then(response => response.json())
+        .then(data => {
+            responseContainer.innerHTML = '';
+            if (data && data.results && data.results.length > 0) {
+                data.results.map(movie => {
+                    console.log(data)
+                    getMovies(movie.title, movie.poster_path, movie.overview, movie.vote_average);
+                })
+            };
+        })
+        .catch(err => console.log(err));
+}
+
+
+getByGenreForm.addEventListener('submit', function (e) {
+    e.preventDefault();
+    const index = selectEl.options.selectedIndex;
+    fetch(`${baseUrl}/discover/movie?api_key=${apiKey}&with_genres=${selectEl.options[index].value}`)
+        .then(response => response.json())
+        .then(data => {
+            responseContainer.innerHTML = '';
+            data.results.map(movie => {
+                getMovies(movie.title, movie.poster_path, movie.overview, movie.vote_average);
+            });
+        })
+        .catch(err => console.log(err));
+})
+
+getByYearForm.addEventListener('submit', function (e) {
+    e.preventDefault();
```

Comment:
Here you can add form validation (actually, would be good to add for other input field as well) to check if the input is a correct year:
```suggestion
    e.preventDefault();
    const yearRegex = /^\d{4}$/;
    if (yearRegex.test(getByYearFormInput.value)) {
```
if the statement above is correct, run the code, otherwise display an error

Reply:
Thank you, I will apply that ASAP.